src/app/services/user-deposit.service.ts

```diff
@@ -26,16 +32,181 @@ export class UserDepositService {
     readonly balance$: Observable<BigNumber>;
     readonly servicesToken$: Observable<UserToken>;
 
+    private readonly userDepositContract$: Observable<Contract>;
+
     constructor(
         private raidenService: RaidenService,
         private tokenInfoRetriever: TokenInfoRetrieverService,
-        private raidenConfig: RaidenConfig
+        private raidenConfig: RaidenConfig,
+        private notificationService: NotificationService
     ) {
-        const userDepositContract$ = this.getUserDepositContractObservable();
-
-        this.balance$ = this.getBalanceObservable(userDepositContract$);
+        this.userDepositContract$ = this.getUserDepositContractObservable();
+        this.balance$ = this.getBalanceObservable(this.userDepositContract$);
         this.servicesToken$ = this.getServicesTokenObservable(
-            userDepositContract$
+            this.userDepositContract$
+        );
+    }
+
+    deposit(amount: BigNumber): Observable<void> {
```

Comment:
All these functions here are very long and tubular/nested. I have a hard time reading through them and don't get lost. I don't know how to improve that much, as especially at the bottom part for each function their are pretty much on a single level of abstraction. But maybe the top parts can be improved?
I hope this kinda made sense. This is just a not about readability. But I generally struggle with long RxJS tubes. But I also think it is easy to abuse them as they one who writes them. Do with this information what ever you want. :grimacing:

Reply:
I can separate the `totalDeposit$` Observable into its own function. That will make the function a little smaller.
I think the notification message objects are quite big and make the pipe less readable, but I don't know how to improve that. As you said it's just one level of abstraction.